x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/nio/SSLOutboundBuffer.java

```diff
@@ -62,10 +59,7 @@ FlushOperation buildNetworkFlushOperation(BiConsumer<Void, Exception> listener)
             byteBuffers[i] = page.byteBuffer();
         }
 
-        return new FlushOperation(byteBuffers, (r, e) -> {
-            IOUtils.closeWhileHandlingException(pagesToClose);
-            listener.accept(r, e);
-        });
+        return new FlushOperation(byteBuffers, (r, e) -> IOUtils.closeWhileHandlingException(pagesToClose));
```

Comment:
`IOUtils.closeWhileHandlingException` seems wrong, shouldn't we be stricter with exceptions here too? Also in `TcpReadWriteHandler` and `MockTcpReadWriteHandler`.

Reply:
Fixed this spot, but I would suggest handling the changes to the `TcpReadWriteHandler`s in a follow-up. Looking at those there's a bunch of closing where all the exceptions are suppressed (e.g. the `InboundPipeline` close which itself suppresses everything on closing its internal resources) so that would be a larger unrelated change to fix.